### PR TITLE
Upgrade dry-system to 0.10.x

### PR DESCRIPTION
**Pull request by arielvalentin**
As the title describes this PR updates the dependency on `dry-system` to version 0.10.x. 

In addition to that, I have added a change to unset the constant when Rails is reloaded to silence `warning: already initialized constant: Container`.

### CODE REVIEW

**Comment by arielvalentin** [comment]
@solnic Thank you so much for accepting my request to collaborate!

What are the current working agreements for PRs? 
E.g. Do we require at least one other core-contributor to approve before merging or am I able to merge my own PRs? 
Is there a minimum time the PR must stay open to allow for others to review? 

Thanks,
Ariel

**Comment by solnic** [comment]
@arielvalentin thank *you* for helping out 😄 

> E.g. Do we require at least one other core-contributor to approve before merging or am I able to merge my own PRs?

Typically, a PR should be accepted by one of the core devs, however, given that this project does not have a maintainer, and you actually use it too, it's fine to merge PRs on your own if *you feel confident about a change*. In case of any doubts, just ping me and I'll help.

> Is there a minimum time the PR must stay open to allow for others to review?

Not really

File: lib/dry/system/rails/version.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module Dry
   module System
     module Rails
-      VERSION = '0.0.2'.freeze
+      VERSION = '0.1.0'.freeze
```
Comment (solnic) [inline lib/dry/system/rails/version.rb:4]: Please make this a separate commit

Comment (arielvalentin) [reply lib/dry/system/rails/version.rb:4]: Sounds good.

**Comment by arielvalentin** [comment]
@solnic I have split the version bump into another commit. Please ♻️ at your earliest convenience.

**Comment by solnic** [comment]
@arielvalentin merged :) if you want that released, I can add you to rubygems's owners too.

**Comment by arielvalentin** [comment]
> @arielvalentin merged :) if you want that released, I can add you to rubygems's owners too.

🎉 Thank you! Yes I would like to release this gem and would be happy to publish it to rubygems.

**Comment by solnic** [comment]
@arielvalentin I added you as the gem owner. If you want to push a new release, make sure to update the CHANGELOG and add a new release on GitHub too.

**Comment by arielvalentin** [comment]
@solnic Will edits to the CHANGELOG require a PR?

**Comment by flash-gordon** [comment]
@arielvalentin nope, feel free to push to master
